savaslabs/savaslabs.github.io
# Issue #3000: add components for colors, links, logos, header and footer

@oksana-c This covers quite a bit!

- [3127](https://pm.savaslabs.com/issues/3127) - Adds a color component, including variable names, hex and rgb codes. I also updated color variables throughout the code. After MVP, I'd like to add other colors and notes on how we're using them all.
- [3131](https://pm.savaslabs.com/issues/3131) - Adds basic link styles. We may need more of these down the road.
- [3002](https://pm.savaslabs.com/issues/3002) - Adds the new SVG logos to the repo
- [3138](https://pm.savaslabs.com/issues/3138) - Adds header and footer menus and menu item components, including template files for the header menu and footer menu
- [3145](https://pm.savaslabs.com/issues/3145) - Adds header region component, including a template file
- [3146](https://pm.savaslabs.com/issues/3146) - Adds footer region component, including a template file

I also made some adjustments to the gulp tasks to make the `gulp serve` process watch files and reload the browser more effectively.

Finally, as I am updating components, I am deleting old SCSS files that are no longer needed. Eventually, all the ones marked to be removed in `main.scss` will be gone.

## To test

- Pull this branch and run `gulp serve`
- Check out the color, link, and logo components
- Check out the header menu link and header menu components. Under the Regions menu item in the style guide, you can see the full header. One shortcoming of the style guide design is that you can't see the header at the full width it'll be on the website. However, you can just look at your local site to see the new header! Also, you won't be able to tell, but the header is transparent so you'd be able to see a hero image beneath it.
- Check out the footer menu link and footer menu components. Under the Regions menu item in the style guide you can see the full footer, or check it out on your local site. One thing I couldn't quite get is the background color on the leftmost part of the footer, which should be light teal. This will take some CSS magic and I ran out of time :) I can definitely take the time to work this out if we feel we have the time.

## Questions for you

- Is the header white or eggshell? I couldn't tell from my copy of the mockup in Sketch.
- The footer spacing is slightly different from the mockup, but I've tried to do everything very consistently by using variables for padding shared by the header and footer so that all the spacing is relative to that base number. If there is anything that needs adjustment please let me know.
- Should we go ahead and make the footer background color exactly as it appears in the mockup? I think it'll take me another hour. I have another .5 hours left in the footer region issue, so we'd only be going over by .5 hours (unless there is more work to be done on the footer per your feedback, which I'm expecting!)
- Does the Roboto Condensed text look overly-bold on your local site? It looks much bolder on my site than it does in your mockups - are you using a lighter weight? I'm using 400 - do I need to be using 300?

## Final thoughts

I will try to make my PRs more discrete from now on! I know this is a lot to review :)

Feedback very welcome!
